lot.php

```diff
@@ -0,0 +1,76 @@
+<?php
+  require_once('helpers.php');
+  require_once('functions.php');
+
+  $is_auth = rand(0, 1);
+  
+  $user_name = 'Павел';
+
+  
+  $connect = db_connect('localhost', 'root', 'root', 'yeticave');
+
+  if (isset($_GET['id'])) {
+    $id = $_GET['id'];
+    if ($id == '') {
+      header("Location: http://243436-yeticave-12/pages/404.html");
+    }
+  } else {
+    header("Location: http://243436-yeticave-12/pages/404.html");
+  }
```

Comment:
Условие стоит упростить (или усложнить, тут как посмотреть), совместив все проверки в один if, чтобы не дублировать код отправки на 404 страницу. Я бы сделал как-то так:
```suggestion
  if (!isset($_GET['id']) || 0 <= $id = (int)$_GET['id']) {
      header("Location: pages/404.html");
  }
```
Также, обрати внимание, что стоит делать редирект на относительный url, так как имя хоста может отличаться в зависимости от настроек сервера.